modules/sr/robot/robot.py

```diff
@@ -87,6 +87,11 @@ def webots_step_and_should_continue(self, duration_ms: int) -> bool:
         Webots telling us whether or not the simulation is about to end).
         """
 
+        if duration_ms <= 0:
+            raise ValueError(
+                "Duration must be greater than zero, not {!r}".format(duration_ms),
+            )
+
```

Comment:
Perhaps if it’s exactly 0 we exit early?

Reply:
I'd rather not -- that has impacts on the simulation (that it can get stuck) which this PR is intending to resolve.